catalog/imaging/catalog-video-transformer/src/main/java/com/connexta/transformer/video/MpegTsInputTransformer.java

```diff
@@ -0,0 +1,200 @@
+
+        this.innerTransformer = inputTransformer;
+        this.metacardTypes = metacardTypes;
+        this.stanag4609Processor = stanag4609Processor;
+        this.klvHandlerFactory = klvHandlerFactory;
+        this.stanagParserFactory = stanagParserFactory;
+        this.defaultKlvHandler = defaultKlvHandler;
+        this.klvProcessor = klvProcessor;
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream)
+            throws IOException, CatalogTransformerException {
+        return transform(inputStream, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream inputStream, final String id)
+            throws IOException, CatalogTransformerException {
+
+        LOGGER.info("processing video input for id = {}", id);
+
+        final MetacardImpl metacard = initializeMetacard(id);
+
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(
+                FILE_THRESHOLD)) {
+
+            populateFileBackedOutputStream(inputStream, fileBackedOutputStream);
+
+            extractInnerTransformerMetadata(id, metacard, fileBackedOutputStream);
+
+            extractStanag4609Metadata(metacard, fileBackedOutputStream);
+
+        }
+
+        return metacard;
+    }
+
+    private MetacardImpl initializeMetacard(String id) throws CatalogTransformerException {
+        final MetacardImpl metacard = new MetacardImpl(this.metacardTypes.stream()
+                .findFirst()
+                .orElseThrow(() -> new CatalogTransformerException(
+                        "no matching metacard type found! id = " + id)));
+
+        metacard.setId(id);
+        metacard.setContentTypeName(CONTENT_TYPE);
+
+        return metacard;
+    }
+
+    private void populateFileBackedOutputStream(InputStream inputStream,
+            FileBackedOutputStream fbos) throws CatalogTransformerException {
+        try {
+            int c = IOUtils.copy(inputStream, fbos);
+            LOGGER.debug("copied {} bytes from input stream to file backed output stream", c);
+        } catch (IOException e) {
+            throw new CatalogTransformerException("Could not copy bytes of content message.", e);
+        }
+    }
+
+    /**
+     * Call the inner transformer with the content data and add every metacard attribute to
+     * the main metacard except for the content type attribute.
+     *
+     * @param id
+     * @param metacard
+     * @param fileBackedOutputStream
+     * @throws IOException
+     * @throws CatalogTransformerException
+     */
+    private void extractInnerTransformerMetadata(String id, MetacardImpl metacard,
+            FileBackedOutputStream fileBackedOutputStream)
+            throws IOException, CatalogTransformerException {
+
+        try (InputStream inputStream = fileBackedOutputStream.asByteSource()
+                .openStream()) {
+
+            Metacard innerMetacard = innerTransformer.transform(inputStream, id);
+
+            innerMetacard.getMetacardType()
```

Comment:
I think you could do this in a simpler way (see [this code](https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java#L152)).
```
metacard = new MetacardImpl(innerMetacard, metacard.getMetacardType());
// reset the content type
metacard.setContentTypeName(CONTENT_TYPE);
```
Of course, you can't modify the metacard you passed in this way so this function would have to return a new metacard. I think that will make some of the other code simpler. You could construct your metacard here rather than in a separate `initializeMetacard()` function.